alternator/expression_factory.cc

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2025-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#include "expressions.hh"
+#include "utils/log.hh"
+
+#include <boost/intrusive/list.hpp>
+#include <absl/container/node_hash_map.h>
+#include <optional>
+
+logging::logger cache_logger("alternator-cache");
+
+namespace alternator {
+namespace {
+
+template<typename Value>
+class lru_string_map {
```

Comment:
I'll need to review this class in detail later, so I won't finish my review right now.
A quick grep yielded that we already have utils/lru.hh but no comments explaining what it is or whether we can reuse it :-(

I wonder if it wouldn't have easier for you and the readers if you just used STL's std::unordered_map plus std::list. It seems lang/wasm_instance_cache.hh does LRU this way.
And utils/loading_cache.hh does LRU with a simple pointer.
It seems everybody invented their own LRU :-)

But each one needs to be reviewed, and to be honest - also needs a unit test (which you're missing).